### DISCUSSION TITLE
How to view LorA training log files? (to check the loss values)

### DISCUSSION THREAD

**Discussion by rambda** [bmaltais/kohya_ss#391] [Q&A]
I know terminal prints epoch loss values, but I accidentally closed it.

According to https://github.com/bmaltais/kohya_ss/blob/master/fine_tune_README.md

    To check the log with TensorBoard, open another command prompt and enter the following in the working folder (I think tensorboard is installed when Diffusers is installed, but if it is not installed, pip install Please put it in tensorboard).

    tensorboard --logdir=logs

But where is the working dir? I tried many dirs inside and outside venv, the logs just never showed up.

Or is there a better way to view the loss values?

**Comment by quasiblob** [comment]
I barely know this stuff, but here it goes.

You can define log folder in the same window where you define your LoRA training image folder.

When training starts, there will be a folder created under your defined log folder, something like this:

\log\20230315212352\network_train\

Open another terminal and start venv in Kohya GUI's install folder root (for PowerShell or the new Windows Terminal):
.\venv\Scripts\activate

Then start the TensorBoard:
tensorboard.exe --logdir "C:\your\dataset\path\here\loradir\log\"

See the printout in terminal view:
_TensorBoard 2.10.1 at http://localhost:6006/ (Press CTRL+C to quit)_

Now control left mouse click the port number to open your web browser. 
Your web browser should open with Tensorboard tab and it should show the data from your log folder.

**Comment by bmaltais** [comment]
Just click the start tensorboard button under the train button. It will open the log in the browser.

**Reply by quasiblob** [reply]
Oh - wait - there is a button now :D ? Was there before?

**Reply by bmaltais** [reply]
New from a few releases back

**Comment by Nightreaver** [comment]
Is it possible to start tensorboard by default without pushing this button?

**Comment by Yosoy-Ed** [comment]
How should we read these logs? What should we check and analyze to improve and iterate future trainings?

**Comment by Nightreaver** [comment]
been trying to figure out same, but found this today:
https://civitai.com/articles/83/using-tensorboard-to-analyze-training-data-and-create-better-models
